src/java_bytecode/java_object_factory.cpp

```diff
@@ -345,22 +365,24 @@ void java_object_factoryt::gen_nondet_init(
 #endif
 
         gen_nondet_init(
+          assignments,
           me,
           _is_sub,
           class_identifier,
-          loc,
           create_dynamic_objects);
       }
     }
     recursion_set.erase(struct_tag);
   }
   else
   {
-    side_effect_expr_nondett se=side_effect_expr_nondett(type);
+    exprt rhs=type.id()==ID_c_bool?
```

Comment:
The code this replaces is:
`
  else if(type.id()==ID_c_bool)
  {
    // We force this to 0 and 1 and won't consider
    // other values.
    return get_nondet_bool(type);
}
`
so I think it's fine